libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1115,7 +1115,16 @@ string ABIFunctions::abiDecodingFunction(Type const& _type, bool _fromMemory, bo
 			return abiDecodingFunctionArray(*arrayType, _fromMemory);
 	}
 	else if (auto const* structType = dynamic_cast<StructType const*>(decodingType.get()))
-		return abiDecodingFunctionStruct(*structType, _fromMemory);
+	{
+		if (structType->dataStoredIn(DataLocation::CallData))
+		{
+			solAssert(!_fromMemory, "");
```

Comment:
Can it happen that `dataStoredIn(DataLocation::Memory)` is true, but `_fromMemory` is false?